src/selectors/fridge.js

```diff
@@ -129,3 +163,72 @@ export const selectTimestampFormatter = createSelector(
     return getTickFormat(date);
   }
 );
+
+export const selectNumberOfHotConsecutiveBreaches = createSelector([selectBreaches], breaches => {
+  const hotBreaches = breaches?.filtered("type == 'HOT_CONSECUTIVE'");
+  return hotBreaches.length;
+});
+
+export const selectNumberOfColdConsecutiveBreaches = createSelector([selectBreaches], breaches => {
+  const coldBreaches = breaches?.filtered("type == 'COLD_CONSECUTIVE'");
+  return coldBreaches.length;
+});
+
+const formatTime = sum => {
+  const asMoment = moment.duration(sum);
+
+  const asDays = Math.floor(asMoment.asDays());
+  const asHours = Math.floor(asMoment.asHours());
+  const asMinutes = Math.floor(asMoment.asMinutes());
+
+  const lengthOfHours = String(asHours).length;
+  const lengthOfMinutes = String(asMinutes).length;
+
+  const addSuffix = (amount, suffix) => `${amount} ${suffix}`;
+
+  if (lengthOfMinutes < 3) return addSuffix(asMinutes, 'm');
+  if (lengthOfHours < 3) return addSuffix(asHours, 'h');
+
+  return addSuffix(asDays, 'd');
+};
+
+export const selectHotCumulativeBreach = createSelector(
+  [selectSelectedFridge, selectFridgeTemperatureLogsFromDate],
+  (fridge, logs) => {
+    const { hotCumulativeBreachConfig } = fridge;
+
+    const { minimumTemperature, duration } = hotCumulativeBreachConfig;
+
+    const logsOverThreshold = logs.filtered('temperature >= $0', minimumTemperature);
+    const sum = logsOverThreshold.sum('logInterval') ?? 0;
+
+    return sum >= duration ? formatTime(sum * 1000) : null;
+  }
+);
+
+export const selectColdCumulativeBreach = createSelector(
+  [selectSelectedFridge, selectFridgeTemperatureLogsFromDate],
+  (fridge, logs) => {
+    const { coldCumulativeBreachConfig } = fridge;
+
+    const { maximumTemperature, duration } = coldCumulativeBreachConfig;
+
+    const logsOverThreshold = logs.filtered('temperature <= $0', maximumTemperature);
+    const sum = logsOverThreshold.sum('logInterval') ?? 0;
+
+    return sum >= duration ? formatTime(sum * 1000) : null;
```

Comment:
Is it worth taking into account the condition where `sum === 0` and `duration` is also `0` due to a bad configuration? the result is a display of a breach which is `0 m`
Technically correct, I suppose, as the duration is 0

Reply:
Yeah I think so. Could also consider having a minimum duration of (1), the same as the logging interval?

Reply:
yeah, that's sensible. can't see why we should support a duration of `0`